src/services/expensesServices.js

```diff
@@ -0,0 +1,108 @@
+const getId = () => {
+  return Math.floor(Math.random() * 1000);
+};
+
+const expenses = [
+  {
+    id: 1,
+    userId: 1,
+    spentAt: 55,
+    title: 'kij',
+    amount: 66,
+    category: 'kk',
+    note: 'jh',
+  },
+];
+
+const resetExpenses = () => {
+  expenses.splice(0, expenses.length);
+};
+
+const getExpenses = (userId, categories, from, to) => {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = expenses.filter((expense) => expense.userId === userId);
+  }
+
+  if (categories) {
+    filteredExpenses = expenses.filter(
+      (expense) => expense.category === categories,
+    );
+  }
+
+  if (from || to) {
+    if (from && to) {
+      filteredExpenses = expenses.filter(
+        (expense) => expense.spentAt >= from && expense.spentAt <= to,
+      );
+    } else if (to) {
+      filteredExpenses = expenses.filter((expense) => expense.spentAt <= to);
+    } else {
+      filteredExpenses = expenses.filter((expense) => expense.spentAt >= from);
+    }
+  }
```

Comment:
you can simplify to avoid nesting if statements:
```suggestion
 if (from && to) {
    return expenses.filter(
      (expense) => expense.spentAt >= from && expense.spentAt <= to,
    );
  }

  if (to) {
    return expenses.filter((expense) => expense.spentAt <= to);
  }

  if (from) {
    return expenses.filter((expense) => expense.spentAt >= from);
  }
```